base/osutils.jl

```diff
@@ -1,20 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-"""
-    OS_NAME
+let UNAME = ccall(:jl_get_UNAME, Any, ())
+    # change some `uname -s` values to "friendly" names
+    # for internal usage
```

Comment:
> It doesn't implement a filesystem or other things that it just delegates to win32.

Did I forget `filesystem` that in my list above? My bad. Cygwin also implements symlinks, filename mappings, and permissions using nt as the HAL. It also supports filesystems such as procfs and handles mounting in some cases such as folders.

> No it wouldn't, midipix still goes through win32. Unless you're microsoft and building code into the kernel (like the WSL subsystem), there's no way to avoid that.

While the `nt` kernel interface is almost undocumented, sufficient information is available due to the need to write drivers against the `nt` side that it is possible to avoid it. Afaict from skimming the code in midipix, it skips the win32 layer and calls directly into `nt` for a number of syscalls. (I don't know whether they avoid linking to win32 entirely, or just in specific instances, although I suspect the former may be true).

> because it doesn't interoperate

yes, this is exactly the fatal flaw in the whole WSL project. midipix may not have this shortcoming and thus is still more interesting (imo).